src/cljc/quil/core.cljc

```diff
@@ -4663,14 +4663,29 @@
   "Temporarily set the fill color for the body of this macro.
    The code outside of with-fill form will have the previous fill color set.
 
-   The fill color has to be in a vector!
-   Example: (with-fill [255] ...)
-            (with-fill [10 80 98] ...)"
-  [fill-args & body]
-  `(let [old-fill# (quil.core/current-fill)]
-     (apply quil.core/fill ~fill-args)
-     ~@body
-     (quil.core/fill old-fill#)))
+   A fill argument of nil disables the fill.
+
+   Example: (with-fill 255 ...)
+            (with-fill [10 80 98] ...)
+            (with-fill (quil.core/color 13 27 42) ...)
+            (with-fill nil ...)"
+  [fill & body]
+  `(let [fill# ~fill
+         fill-was-enabled# (.fill (quil.core/current-graphics))
```

Comment:
Did you test it on cljs? Here is the docs how to quickly run cljs example from quil repo directly: https://github.com/quil/quil#clojurescript-1 it allows you to update `core.cljc` and automatically recompile.

Reply:
Actually nevermind, current-fill and current-stroke are not supported on cljs. I'm going to make this macro clj-only later.